### DISCUSSION TITLE
Adding multiple scenes to print multiple lasers at the same time

### DISCUSSION THREAD

**Discussion by serjik69** [meerk40t/meerk40t#2536] [Q&A]
Hi, how to properly add multi-scene support for simultaneous operation of multiple lasers
I changed the files wxmscene.py, elements.py, elements.py
which classes, functions need to be changed

wxmscene.py
def register_panel_scene(window, context):
	panel0 = MeerK40tScenePanel(window, wx.ID_ANY, context=context, index=0)
    control.AddPage(panel0, "scene0")
    panel1 = MeerK40tScenePanel(window, wx.ID_ANY, context=context, index=1)
    control.AddPage(panel1, "scene1")

def on_note_page_change(event=None):
	if control.GetPageText(control.GetSelection()) == "scene0":
        context.kernel.activate_service_path("elements", "elements0")
	else:
        context.kernel.activate_service_path("elements", "elements1")
window.on_pane_create(pane)
context.register("pane/scene", pane)
	
elements.py
# kernel.add_service("elements", Elemental(kernel))
kernel.add_service("elements", Elemental(kernel,0))
kernel.add_service("elements", Elemental(kernel,1))

problem, I can't draw on 1 scene
there's a mistake on stage 2

MeerK40t crash log. Version: 0.9.4000 pkg on Windows: Python 3.12.3: AMD64 - wxPython: 4.2.1 msw (phoenix) wxWidgets 3.2.2.1
Traceback (most recent call last):
  File "C:\Projects\meerk40t\meerk40t\gui\scene\scenepanel.py", line 273, in on_left_mouse_up
    self.scene.event(
  File "C:\Projects\meerk40t\meerk40t\gui\scene\scene.py", line 780, in event
    response = current_widget.event(
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Projects\meerk40t\meerk40t\gui\toolwidgets\toolpointlistbuilder.py", line 117, in event
    self.point_added()
  File "C:\Projects\meerk40t\meerk40t\gui\toolwidgets\toolline.py", line 37, in point_added
    self.end_tool()  # That will call everything
    ^^^^^^^^^^^^^^^
  File "C:\Projects\meerk40t\meerk40t\gui\toolwidgets\toolpointlistbuilder.py", line 189, in end_tool
    followup = self.create_node()
               ^^^^^^^^^^^^^^^^^^
  File "C:\Projects\meerk40t\meerk40t\gui\toolwidgets\toolline.py", line 32, in create_node
    elements.classify([node])
  File "C:\Projects\meerk40t\meerk40t\core\elements\elements.py", line 2586, in classify
    not self.classify_black_as_raster
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'Elemental' object has no attribute 'classify_black_as_raster'

Thank you very much.

**Comment by jpirnay** [comment]
The error you are referring to has recently be fixed in the github main branch (https://github.com/meerk40t/meerk40t/commit/c2ccd1efb25d2f326a75e6ed52e294f32bbb9ea2), so you might want to pull from the main repository to be again in sync.

**Comment by jpirnay** [comment]
I need to look to the development past, tat had once created a stub for multi pane support. There were some issues with it and we had more pressing business at that time, so the development stopped at this point: https://github.com/meerk40t/meerk40t/tree/dual-scenes
